### PR TITLE
deprecated @treelike for @functor with warning

### LINKED ISSUE #1339: deprecate treelike
```julia
using Flux
using Flux: @treelike
struct Cat{T}
    layers::T
end
@treelike Cat
```

> LoadError: UndefVarError: functorm not defined
> in expression starting at In[17]:6
> 
> Stacktrace:
>  [1] @treelike(::LineNumberNode, ::Module, ::Vararg{Any,N} where N) at /home/cossio/.julia/packages/Flux/05b38/src/functor.jl:58
>  [2] include_string(::Function, ::Module, ::String, ::String) at ./loading.jl:1091
>

### PR DESCRIPTION
**Pull request by mungu42**
closes #1339

### PR Checklist

- [ ] Tests are added
- [ ] Entry in NEWS.md
- [ ] Documentation, if applicable
- [ ] Final review from `@dhairyagandhi96` (for API changes).

### CODE REVIEW

File: src/functor.jl [hunk 1]
```diff
@@ -55,7 +55,8 @@ end
 
 # Deprecated stuff
 macro treelike(args...)
-  functorm(args...)
+  @warn "@treelike has been deprecate please use @functor instead"
+  functor(args...)
```
Comment (CarloLucibello) [inline src/functor.jl:59]: `functorm` is gone as well? is `functor` the correct replacement here?

Comment (mungu42) [reply src/functor.jl:59]: seems like it, couldn't find it using global search

Comment (mungu42) [reply src/functor.jl:59]: Currently, #1339 also mentions that `@functor` should be used

Comment (DhairyaLGandhi) [reply src/functor.jl:59]: `functnorm` has been moved to Functors.jl, so using it from there would be an option

**Comment by CarloLucibello** [comment]
I just removed `@treelike` in #1377, so I will close this PR

**Pull request closed without merging**